code/modules/projectiles/guns/ego_gun/aleph.dm

```diff
@@ -66,3 +66,72 @@
 #undef SHOT_MODE
 #undef DOT_MODE
 #undef AOE_MODE
+
+/obj/item/gun/ego_gun/nihil
+	name = "nihil"
+	desc = "Having decided to trust its own intuition, the jester spake the names of everyone it had met on that path with each step it took."
+	icon_state = "nihil"
+	inhand_icon_state = "nihil"
+	ammo_type = /obj/item/ammo_casing/caseless/ego_nihil
+	weapon_weight = WEAPON_HEAVY
+	fire_sound = 'sound/weapons/fixer/generic/energy1.ogg'
+	fire_sound_volume = 50
+	fire_delay = 10
+	attribute_requirements = list(
+							FORTITUDE_ATTRIBUTE = 80,
+							PRUDENCE_ATTRIBUTE = 80,
+							TEMPERANCE_ATTRIBUTE = 100,
+							JUSTICE_ATTRIBUTE = 80
+							)
+	var/wrath
+	var/despair
+	var/greed
+	var/hate
+	var/list/powers = list("hatred", "despair", "greed", "wrath")
+
+/obj/item/gun/ego_gun/nihil/attackby(obj/item/I, mob/living/user, params)
+	..()
+	if(!istype(I, /obj/item/nihil))
+		return
+
+	if(powers[1] == "hatred" && istype(I, /obj/item/nihil/heart))
+		powers[1] = "hearts"
```

Comment:
I guess this way could be done? I was kinda going for an assoc list idea but I think this works well enough.